torch/_refs/__init__.py

```diff
@@ -4783,6 +4824,18 @@ def lerp(start: Tensor, end: Tensor, weight: Union[Tensor, NumberType]):
     stride = utils.compute_elementwise_output_strides(*_maybe_broadcast(*inputs))
     if output.stride() != stride:
         return prims.copy_strided(output, stride)
+
+    device = None
+    from torch._subclasses.fake_tensor import FakeTensor
+
+    for t in inputs:
+        if isinstance(t, FakeTensor):
+            device = t.fake_device
+            break
+
+    if is_noncontiguous_supported(device) is False:
+        output = output.new_empty(output.shape)
```

Comment:
Can you factor this and similar sites into a helper function?

Reply:
done.